docs/client/initializing.md

```diff
@@ -0,0 +1,105 @@
+
+| Library | Description | Min Version
+|------------------|-------------|-------------
+|[`azure-core`][azure_core_library]|The most important library to have installed. It provides shared exceptions and modules for all the Python SDK client libraries.|1.8.2
+|[`msrest`][msrest_library]|Library mainly used for serializing and deserializing objects now|0.6.18
+|[`azure-mgmt-core`][azure_mgmt_core_library]|Required if you're generating mgmt plane code (see `--azure-arm` flag in our [flag index][flag_index]. Provides mgmt plane specific shared exceptions and modules.|1.2.1
+
+> Note: We highly recommend tying your library to a major version, for instance, adding `azure-core<2.0.0` to tie the `azure-core` library to `1.x.x`
+
+## Initializing Your Client
+
+Next, on to initialization. Your constructor can take any number of parameters. If your client has no parameters (no client parameters detailed
+in the swagger (TODO: link to swagger docs) and you choose to generate without credentials), initializing your client would just look like
+
+```python
+from azure.pets import PetsClient
+
+client = PetsClient()
+```
+
+However, by default we generate clients with credentials, so continue on to [Authenticating Your Client](#authenticating-your-client "Authenticating Your Client")
+to find out how to input a credential.
+
+## Authenticating Your Client
+
+By default we generate our clients with an [Azure Active Directory (AAD) token credential][aad_authentication]. We always recommend
+using a [credential type][identity_credentials] obtained from the [`azure-identity`][azure_identity_library] library for AAD authentication. For this example,
+we use the most common [`DefaultAzureCredential`][default_azure_credential].
+
+As an installation note, the [`azure-identity`][azure_identity_library] library is not a requirement in the basic `setup.py` file we generate
+(see `--basic-setup-py` in our [flag index][flag_index] for more information), so you would need to install this library separately.
+
+```python
+from azure.identity import DefaultAzureCredential
+from azure.pets import PetsClient
+
+client = PetsClient(credential=DefaultAzureCredential())
+```
+
+You can also have your generated client take in an [`AzureKeyCredential`][azure_key_credential] instead. To do so, generate with flag `--credential-types=AzureKeyCredential`,
+and for more information on this flag, see our [flag index][flag_index]
+
+```python
+from azure.core.credentials import AzureKeyCredential
+from azure.pets import PetsClient
+
+credential = "myCredential"
+client = PetsClient(credential=AzureKeyCredential(credential))
+```
+
+Currently, we only support generating credentials of type `TokenCredential` and / or `AzureKeyCredential`. If you'd like to use your own custom credential,
+you can still pass it in to the client. However, you may have to use a custom authentication policy to handle the credential. That can also be passed in to the
```

Comment:
or something like that.... I got confused when reading it and wasn't sure what "it" meant right away

```suggestion
you can pass the custom type into the client. However, you may have to use a custom authentication policy to handle the credential. That can also be passed in to the
```